packages/signers/src/lit-protocol/signer.ts

```diff
@@ -0,0 +1,197 @@
+        debug: params.debug ?? false,
+      });
+    this._rpcUrl = params.rpcUrl;
+  }
+  signerType: string = SIGNER_TYPE;
+
+  /**
+   * if generic type is `LitAuthMethod`, authenticates the supplied authentication material.
+   * if type `SessionSigsMap`, this implementation will respect the existing auth and use the session material.
+   * @param props {LITAuthenticateProps} Authentication params, only `context` is required
+   * @returns {LitSessionSigsMap} Authenticated session material
+   */
+  authenticate = async (
+    props: LitAuthenticateProps<C>
+  ): Promise<LitUserMetadata> => {
+    if (!this.session) {
+      await this._doAuthentication(props);
+    }
+    if (!this.session) {
+      throw new Error("Not Authenticated");
+    }
+
+    return this.session;
+  };
+
+  getAuthDetails = async (): Promise<LitUserMetadata> => {
+    this._checkInternals();
+    return this._authContext as LitSessionSigsMap;
+  };
+
+  getAddress = async () => {
+    this._checkInternals();
+    const address = await this.signer?.getAddress();
+
+    return address as `0x${string}`;
+  };
+
+  signMessage = async (msg: Uint8Array | string) => {
+    this._checkInternals();
+
+    return this.signer?.signMessage(msg) as Promise<Address>;
+  };
+
+  signTypedData = (params: SignTypedDataParams) => {
+    this._checkInternals();
+
+    return this.signer?._signTypedData(
+      params.domain as TypedDataDomain,
+      params.types as any,
+      params.message
+    ) as Promise<Address>;
+  };
+
+  private _checkInternals() {
+    if (!this._authContext) {
+      throw new Error("Not Authenticated");
+    }
+
+    if (!this.signer) {
+      throw new Error("Signer is not initalized, did you call authenticate?");
+    }
+  }
+
+  private async _doAuthentication(props: LitAuthenticateProps<C>) {
+    // check if the object is structed as an auth method
+    // if so we sign the session key with the auth method
+    // as the auth material. If a session signature
+    // is provided then we skip this step.
+    if (Object.keys(props.context).indexOf("accessToken") > 0) {
+      const resourceAbilities = [
+        {
+          resource: new LitPKPResource("*"),
+          ability: LitAbility.PKPSigning,
+        },
+      ];
+      const sessionKeypair = props.sessionKeypair || generateSessionKeyPair();
+      const chain = props.chain || "ethereum";
+      let authNeededCallback: any;
+      if (props.context?.authMethodType === 1) {
+        authNeededCallback = async (params: AuthCallbackParams) => {
+          const response = await this.inner.signSessionKey({
+            statement: params.statement,
+            authMethods: [props.context as LitAuthMethod],
+            authSig: JSON.parse(props.context.accessToken as string) as AuthSig,
+            pkpPublicKey: `0x${this._pkpPublicKey}`,
+            expiration: params.expiration,
+            resources: params.resources,
+            chainId: 1,
```

Comment:
should this chain id match the chain of the provider using this signer? if so, should the signer constructor config take in a viem chain object?

Reply:
see above